bitcoin/src/pow.rs

```diff
@@ -913,6 +926,116 @@ impl kani::Arbitrary for U256 {
     }
 }
 
+/// Error from `get_block_header_by_height` callback.
+///
+/// Occurse when there is an error while get a new block header
+/// from an external source.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum GetBlockHeaderError {
+    /// Block Header not found
+    NotFound,
+    /// Generic error with custom message
+    Generic(String),
+}
+
+impl fmt::Display for GetBlockHeaderError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let msg = match self {
+            Self::Generic(error) => error,
+            Self::NotFound => "Block header not found",
+        };
+        write!(f, "{msg}")
```

Comment:
```suggestion
        write!(f, "{}", msg)
```

Reply:
This is the reason CI is breaking. (1.48.0 Rust can't embed variables like this)

Reply:
ops I forget this